runtime/strategies/search-tokens-to-particles.js

```diff
@@ -98,14 +125,28 @@ export class SearchTokensToParticles extends Strategy {
     return [...generated, ...terminal];
   }
 
-  _addParticle(token, particle) {
-    this._particleByToken[token] = this._particleByToken[token] || [];
-    this._particleByToken[token].push(particle);
+  _addThing(token, thing, thingByToken, thingByPhrase) {
+    if (!token) {
+      return;
+    }
+    this._addThingByToken(token.toLowerCase(), thing, thingByToken);
+
+    if (thingByPhrase) {
```

Comment:
thingByPhrase is always present, isn't it?

Reply:
thanks for noticing! done